src/bloaty.cc

```diff
@@ -1386,57 +1389,55 @@ class Bloaty {
   std::vector<ConfiguredDataSource*> sources_;
   std::vector<std::string> source_names_;
 
-  std::vector<std::unique_ptr<ObjectFile>> input_files_;
-  std::vector<std::unique_ptr<ObjectFile>> base_files_;
-  std::map<std::string, std::unique_ptr<ObjectFile>> debug_files_;
+  std::vector<std::pair<std::string, std::string>> input_files_;
+  std::vector<std::pair<std::string, std::string>> base_files_;
+  std::map<std::string, std::string> debug_files_;
 };
 
 Bloaty::Bloaty(const InputFileFactory& factory, const Options& options)
     : file_factory_(factory), options_(options) {
   AddBuiltInSources(data_sources, options);
 }
 
-void Bloaty::AddFilename(const std::string& filename, bool is_base) {
+std::unique_ptr<ObjectFile> Bloaty::GetObjectFile(const std::string& filename,
+                                                  bool allow_web_assembly) const {
   std::unique_ptr<InputFile> file(file_factory_.OpenFile(filename));
   auto object_file = TryOpenELFFile(file);
 
   if (!object_file.get()) {
     object_file = TryOpenMachOFile(file);
   }
 
-  if (!object_file.get()) {
+  if (!object_file.get() && allow_web_assembly) {
```

Comment:
Is there a need to specifically disallow debug files with WebAssembly here? There is a feature request to support DWARF with WebAssembly: https://github.com/google/bloaty/issues/188

Reply:
I was wondering that myself, but I honestly have no idea what a WebAssembly file is, so I was just preserving the previous behavior.

If there is no need for this, then I'm very happy to remove that extra complexity.